File: go/rna-transcription/rna_transcription.go
package strand

func ToRNA(dna string) string {

	nucleotideComplement := map[rune]rune{'G': 'C', 'C': 'G', 'T': 'A', 'A': 'U'}
	rna := ""

	for _, char := range dna {
		_, ok := nucleotideComplement[char]
		if !ok {
			return ""
		}
		rna += string(nucleotideComplement[char])
	}
	return rna
}

File: python/rna-transcription/rna_transcription.py
def to_rna(dna_strand: str) -> str:
    rna_strand = []

    nucleotide_complement = {
    "G": "C",
    "C": "G",
    "T": "A",
    "A": "U"
    }

    for nucleotide in dna_strand:
        if nucleotide in nucleotide_complement:
            rna_strand.append(nucleotide_complement[nucleotide])

    return "".join(rna_strand)
